packages/ui-components/src/components/Inputs/FileUploadField.tsx

```diff
@@ -1,172 +1,244 @@
+  return (
+    <>
+      {value}&nbsp;{unit}
+    </>
+  );
 };
 
 interface FileUploadFieldProps {
   onChange: (
+    files: File[] | FileList | null,
     event: React.ChangeEvent<HTMLInputElement> | null,
-    fileName?: string,
-    files?: FileList | null | undefined,
   ) => void;
   name: string;
-  fileName: string;
   multiple?: boolean;
-  textOnButton?: string;
   label?: string;
   tooltip?: string;
   helperText?: string;
-  showFileSize?: boolean;
   maxSizeInBytes?: number;
   FormHelperTextComponent?: React.ElementType;
   required?: boolean;
-  buttonVariant?: 'text' | 'outlined' | 'contained';
   accept?: string;
 }
 
 export const FileUploadField = ({
-  onChange = () => {},
-  name,
-  fileName,
-  multiple = false,
-  textOnButton,
-  label,
-  tooltip,
+  onChange,
+  FormHelperTextComponent = 'p',
+  accept,
   helperText,
-  showFileSize = false,
+  label,
   maxSizeInBytes,
-  FormHelperTextComponent = 'p',
-  required,
-  buttonVariant = 'contained',
-  accept = '*',
+  multiple = false,
+  name,
+  required = false,
+  tooltip,
 }: FileUploadFieldProps) => {
-  const inputEl = useRef<HTMLInputElement | null>(null);
-  const text = textOnButton || `Choose file${multiple ? 's' : ''}`;
-
-  const [error, setError] = useState<string | null>(null);
-  const [sizeInBytes, setSizeInBytes] = useState<number | null>(null);
-
-  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    let newName;
-    const input = inputEl.current;
-
-    if (!input || !input.files) return;
-
-    if (input.files.length > 1) {
-      newName = `${input.files.length} files selected`;
-    } else {
-      newName = event.target.value.split('\\').pop();
-    }
-
-    if (maxSizeInBytes) {
-      for (const file of Array.from(input.files)) {
-        const { size: newSizeInBytes } = file;
-        if (newSizeInBytes > maxSizeInBytes) {
-          setSizeInBytes(null);
-          setError(
-            `Error: file is too large: ${humanFileSize(
-              newSizeInBytes,
-            )}. Max file size: ${humanFileSize(maxSizeInBytes)}`,
-          );
-          onChange(event, undefined, null);
-          return;
-        }
-      }
-    }
-
-    if (multiple) {
-      onChange(event, newName, input.files);
-      // We don't support file size label if multiple
-    } else {
-      const [file] = Array.from(input.files);
-      setSizeInBytes(file.size);
-      onChange(event, newName, input.files);
-    }
-    setError(null);
+  const [files, setFiles] = useState<File[]>([]);
+  const hasFileSelected = files.length > 0;
+
+  const onDropAccepted = (acceptedFiles: File[], event: DropEvent) => {
+    const deduped = acceptedFiles.filter(
+      acceptedFile =>
+        !files.map(file => JSON.stringify(file)).includes(JSON.stringify(acceptedFile)),
+    );
+    setFiles(files.concat(deduped));
+    console.log(files);
+    onChange(files, event as React.ChangeEvent<HTMLInputElement>);
```

Comment:
Out of interest, is there any particular reason to manage files in state AND from the parent? Do you foresee any bugs managing both?

Reply:
I’m irked by it, to be honest. The only reason this component manages state is so that files can be removed. (My approach doesn’t use `inputRef.current.value`.)

I started off with state management only in the parent (i.e. existing behaviour), and just passed the `acceptedFiles` up to the parent, but the `react-dropzone` API has no way to remove or clear files (except by making a new file selection). Had a word with Felicity and her preference is to be able to add/remove files individually. Rationale being that `<input>`’s normal behaviour of having to do multi-selects “is one go” isn’t super obvious to people with lower technical self-efficacy

Not so worried about bugs, but I do think this is awkward to trace/debug. If you have ideas I’d love to hear 😅

Reply:
Hmm, that's annoying. So doing `onChange` with a filtered array of files won't remove the file from the list?

If that doesn't work, you could also just do a `useEffect` on change of files array (probably of stringified file names array or something, so it doesn't re-render too many times) and call `onChange` there so that you don't have to remember to add it to he various on change handlers inside this component. Also adding some comments, so someone (me) doesn't inadvertently break it later 😅

Reply:
Ah, the trouble is you can add and remove files individually. So when `multiple=true`, you don’t _have_ to ⌘-click or ⇧-click to select multiple files all at once. You can drop some files, then drop some more files, which is why the `setState` call concatenates the new selection to the existing selection. It doesn’t just replace it

The difficulty with using only file name (and the reason which I stringify the entire `File` object) is that there is an edge case where you might select two files with the same name from different folders. The browser doesn’t reveal a fully qualified path to us. We just get the basename, so it isn’t actually guaranteed to be unique 😵

In fact, even JSON.stringify(file) isn’t guaranteed to work perfectly, but I just figured it’s **extremely** unlikely for two non-identical files to have the same name **and** filesize

`useEffect` seems like a good idea regardless—I’ll give that a go!

Reply:
I ended up deciding against `useEffect` because:

1. `setFile` is called in only one place
2. the way React queues updates could cause it to run later than expected

Using the change handler is [how the React docs advise notifying the parent about state changes](https://react.dev/learn/you-might-not-need-an-effect#notifying-parent-components-about-state-changes)